homeworkOne-todoapp/app.js

```diff
@@ -0,0 +1,75 @@
+var Todo = /** @class */ (function () {
+    function Todo(taskName, endDate, completed, description) {
+        this.taskName = taskName;
+        this.endDate = endDate;
+        this.completed = completed;
+        this.description = description;
+        this.taskName = taskName;
+        this.endDate = endDate;
+        this.completed = false;
+        this.description = description;
+    }
+    Todo.prototype.finish = function () {
+        this.completed = true;
+    };
+    Todo.prototype.modify = function (taskName, endDate, description) {
+        this.taskName = taskName;
+        this.endDate = endDate;
+        this.description = description;
+    };
+    return Todo;
+}());
+var TodoList = /** @class */ (function () {
+    function TodoList() {
+        this.allTodos = new Array;
+    }
+    //functie creare task
+    TodoList.prototype.createTodo = function (taskName, endDate, description) {
+        var newItem = new Todo(taskName, endDate, false, description);
+        this.allTodos.push(newItem);
+    };
+    //functie listare taskuri
+    TodoList.prototype.listTodos = function () {
+        console.table(this.allTodos);
+    };
+    //functie modificare task
+    TodoList.prototype.modifyTask = function (index, taskName, endDate, description) {
+        this.allTodos[index].taskName = taskName || this.allTodos[index].taskName;
+        this.allTodos[index].endDate = endDate || this.allTodos[index].endDate;
+        this.allTodos[index].description = description || this.allTodos[index].description;
+        this.listTodos();
+    };
+    //functie finalizare task    
+    TodoList.prototype.finishTask = function (index) {
+        this.allTodos[index].finish();
+        this.listTodos();
+    };
+    //functie pentru stergerea unui task
+    TodoList.prototype.deleteTask = function (index) {
+        this.allTodos.splice(index, 1);
+        this.listTodos();
+    };
+    //functie listare taskuri nefinalizate
+    TodoList.prototype.listUnfinishedTodos = function () {
+        var finished = this.allTodos.filter(function (t) { return !t.completed; });
+        console.table(finished);
+        this.listTodos();
```

Comment:
Metoda ta face mai mult decat scrie in definitia ei :)

Reply:
Phew. Bine ca nu facea mai putin :D Am rezolvat!